app/assets/javascripts/components/overview/my_articles.jsx

```diff
@@ -26,27 +26,38 @@ export const MyArticles = createReactClass({
   },
 
   render() {
-    const assignOptions = { user_id: this.props.current_user.id, role: 0 };
-    const reviewOptions = { user_id: this.props.current_user.id, role: 1 };
+    const { assignments, current_user } = this.props;
+    const user_id = current_user.id;
+    const assignOptions = { user_id, role: 0 };
+    const reviewOptions = { user_id, role: 1 };
 
-    const assigned = getFiltered(this.props.assignments, assignOptions);
-    const reviewing = getFiltered(this.props.assignments, reviewOptions);
-    const allAssignments = assigned.concat(reviewing);
-    const assignmentCount = allAssignments.length;
-    const assignments = allAssignments.map((assignment, i) => {
-      return (
-        <MyAssignment
-          key={assignment.id}
-          assignment={assignment}
-          course={this.props.course}
-          username={this.props.current_user.username}
-          last={i === assignmentCount - 1}
-          current_user={this.props.current_user}
-          wikidataLabels={this.props.wikidataLabels}
-        />
-      );
+    const assignmentsWithSandboxes = assignments.map((article) => {
+      if (article.role === 0) return article;
+      const related = assignments.find((assignment) => {
+        return assignment.article_id === article.article_id
+          && assignment.user_id
```

Comment:
Remove this?

Reply:
I remember what this was for. If we don't include this check, it will find the first assignment even if the `user_id` is `undefined`.

Reply:
Ah. Carry on, then. This will get replaced with assignment status at some point soon-ish anyway.